Disable broken test for libxml2 xmlReadMemory
It's not defined in xmlreader.h (it's defined in parser.h).
Since we never install a 32-bit version and the test fails, we disable it.
--- a/llvm/cmake/config-ix.cmake  2023-01-19 11:37:00.000000000 -0800
+++ b/llvm/cmake/config-ix.cmake 2023-01-19 11:37:06.000000000 -0800
@@ -157,20 +157,10 @@
   elseif(NOT LLVM_USE_SANITIZER MATCHES "Memory.*")
     find_package(LibXml2)
   endif()
   if(LibXml2_FOUND)
-    # Check if libxml2 we found is usable; for example, we may have found a 32-bit
-    # library on a 64-bit system which would result in a link-time failure.
-    cmake_push_check_state()
-    list(APPEND CMAKE_REQUIRED_INCLUDES ${LIBXML2_INCLUDE_DIRS})
-    list(APPEND CMAKE_REQUIRED_LIBRARIES ${LIBXML2_LIBRARIES})
-    list(APPEND CMAKE_REQUIRED_DEFINITIONS ${LIBXML2_DEFINITIONS})
-    check_symbol_exists(xmlReadMemory libxml/xmlreader.h HAVE_LIBXML2)
-    cmake_pop_check_state()
-    if(LLVM_ENABLE_LIBXML2 STREQUAL FORCE_ON AND NOT HAVE_LIBXML2)
-      message(FATAL_ERROR "Failed to configure libxml2")
-    endif()
+    set(HAVE_LIBXML2 ON)
   endif()
   set(LLVM_ENABLE_LIBXML2 "${HAVE_LIBXML2}")
 endif()
 
 if(LLVM_ENABLE_CURL)
